### DISCUSSION TITLE
How to check if on page 1 (not page 2, 3, etc) when rendering a block

### DISCUSSION THREAD

**Discussion by realthemes** [WordPress/gutenberg#65684] [Q&A] [answered]
I'm working on rendering a custom block in WordPress and need to conditionally display it only on the first page (page 1) of a paginated query or on the front page. The block should not appear on subsequent pages like page 2, 3, etc.

I've tried using ```get_query_var('paged')``` and ```get_query_var('page')```, but neither seems to work in my case. I also tried using ```global $post```, but it returns 1 all the time, regardless of the page I'm on, and unfortunately, the block still shows on paginated pages (e.g., /?query-1-page=2).

Is there a reliable way to check if I am on page 1 (or the first page of a paginated query) when rendering a block? Please help me.

PS I'm using my custom block theme with custom Gutenberg blocks

**Comment by pshemek** [comment, marked as answer]
Hi, looking at the way the Post Template block (https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/post-template/index.php) is rendered, I think you can use the “render_block” filter for your block type and derive the query context in a similar way this block does.

**Reply by realthemes** [reply]
Yes, I found with the help of the ```pagination-numbers``` block that ```$_GET["query-X-page"]``` returns the page number. The only problem is how to pass the ```queryId``` to my block, since my block isn't inside a ```query loop```, and therefore doesn't get the ```queryId``` ```context```. I configured my block to loop through all parameters in ```$_GET``` and detect the page number. Thank you! I will mark your answer, as the method using ```$_GET``` is utilized by the WordPress team, and I will use it for my block as well.
